### PR TITLE
FS-3036-bug-fix-map-application-q-a-for-text-file

### PR DESCRIPTION
**Pull request by ramsharma-prog**
https://dluhcdigital.atlassian.net/jira/software/c/projects/FS/boards/50?view=detail&selectedIssue=FS-3036

### Description
This PR addresses several improvements:
- Bug fixes for the funding required form mapping and formatting in a text file. 
- It enhances the formatting of date-time year values.
- Removes null values from strings, list. 
- Updates the radio button's answer and strings by removing the "-" character, excluding those that are part of an HTTP or HTTPS link.

### Test
- [X] Unit tests and other appropriate tests added or updated

### Screenshot
Screenshot 2023-07-07 at 19 13 01

Screenshot 2023-07-07 at 19 13 27

Screenshot 2023-07-07 at 19 19 35

Screenshot 2023-07-07 at 19 18 38

### CODE REVIEW

**Review by rayhan15243 — APPROVED** [review]

**Review by gio-karunakaran — APPROVED** [review]
LGTM

File: fsd_utils/mapping/application/application_utils.py [hunk 1]
```diff
@@ -40,12 +41,24 @@ def format_answer(answer):
         if "null" in answer:
             return re.sub(r"\s*null\s*,?", "", answer)
 
+        if isinstance(answer, str):
+            if answer.startswith("http") or answer.startswith("https"):
+                return answer
+            elif "-" in answer:
```
Comment (srh-sloan) [inline fsd_utils/mapping/application/application_utils.py:47]: Do we always want to replace all hyphens with spaces? The free text ones could have a valid `-` character. Wouldn't we only remove the hyphens for like bullet/radio button answers where the 'answer' is actually a hard coded backend value?

Comment (srh-sloan) [reply fsd_utils/mapping/application/application_utils.py:47]: So maybe do this conditionally based on field type?

File: tests/test_data_utils.py [hunk 2]
```diff
@@ -87,14 +87,14 @@ def test_get_remote_data_json_404(self, flask_test_client):
...
-            "expected_response": "- Sky builders\n     - trust builders",
+            "expected_response": ". Sky builders\n     . trust builders",
         },
         "integer_values": {
             "input_data": [
                 {"GLQlOh": "cost one", "JtwkMy": 4444},
                 {"GLQl6y": "cost two", "JtwkMt": 4455},
             ],
-            "expected_response": "- cost one: 4444\n     - cost two: 4455",
+            "expected_response": ". cost one: 4444\n     . cost two: 4455",
```
Comment (srh-sloan) [inline tests/test_data_utils.py:97]: Don't know if it's part of the requirements, but shouldn't we have a £ sign in here if it's currency?

**Review by srh-sloan — COMMENTED** [review]
Couple of queries around the format but otherwise good 👍 
+1 for tests!

File: tests/test_data_utils.py (hunk 2 above)
Comment (ramsharma-prog) [reply tests/test_data_utils.py:97]: The formatting isn't field specific so adding £ sign could add a prefix to the string but defo worth looking into it

File: fsd_utils/mapping/application/application_utils.py (hunk 1 above)
Comment (ramsharma-prog) [reply fsd_utils/mapping/application/application_utils.py:47]: I thought of it too but the radio button doesn't seem to have a specific field type so it was kind of tricky. I will defo look into it or should we leave the hyphens as it is for the checkbox, radio fields, etc. It does handle the HTTP response

**Pull request merged**